nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -1027,19 +1028,24 @@ public boolean accept(final File dir, final String filename) {
         cmd.add("-Dorg.apache.nifi.bootstrap.config.log.dir=" + nifiLogDir);
         cmd.add("org.apache.nifi.NiFi");
         if (props.containsKey(NIFI_BOOTSTRAP_SENSITIVE_KEY) && !StringUtils.isBlank(props.get(NIFI_BOOTSTRAP_SENSITIVE_KEY))) {
-            cmd.add("-k " + props.get(NIFI_BOOTSTRAP_SENSITIVE_KEY));
+            File password_file = new File(confDir, "sensitive.key");
+
+            // Restrict permissions:
+            password_file.setReadable(false,false);
```

Comment:
Both of `RunNifi` and `Nifi` are run as the `run.as` user, referring to `nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/bin/nifi.sh`. `setOwner` is therefore implicitly already the `run.as` user---we can make it explicit, no trouble, but what do we gain from it?